File: src/main/java/scraping/query/QuerySchool.java
package scraping.query;

import nyu.Term;
import org.asynchttpclient.Request;
import org.asynchttpclient.RequestBuilder;
import org.asynchttpclient.uri.Uri;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.concurrent.ExecutionException;

public final class QuerySchool {

  private static final Logger logger =
      LoggerFactory.getLogger("scraping.query.QuerySchool");
  private static final Uri ROOT_URI =
          Uri.create("https://m.albert.nyu.edu/app/catalog/classSearch");

  public static String querySchool(Term term) {
    logger.info("querying school for term={}", term);

    Request request = new RequestBuilder()
            .setUri(ROOT_URI)
                              .setRequestTimeout(60000)
                              .build();

    try {
      return GetClient.getClient().executeRequest(request).get().getResponseBody();
    } catch (InterruptedException | ExecutionException e) {
      logger.error("Error (term=" + term + "): " + e.getMessage());
      return null;
    }
  }
}

File: src/main/kotlin/services/query_school.kt
package services

import com.github.kittinunf.fuel.Fuel
import nyu.Term
import mu.KotlinLogging
import java.util.concurrent.CompletableFuture
import java.util.concurrent.Future

private val queryLogger = KotlinLogging.logger("services.query_catalog")
private const val ROOT_URL = "https://m.albert.nyu.edu/app/catalog/classSearch"
/**
 * The meat of querying the catalog resides here.
 */
fun querySchool(term: Term): String {
    queryLogger.info { "querying school for term=$term" }
    val request = Fuel.get("${ROOT_URL}/${term.id}")
    return request.response().toString()
}

